proposals/NNNN-sdl-js-pwa.md

```diff
@@ -0,0 +1,213 @@
+  3. For local apps
+    1. "cloudTransportType" will be "ws" (WebSocket) or "wss" (WebSocket-Secure)
+    2. "endpoint" parameter will be omitted. Instead the HMI is responsible to launch local apps.
+  4. If Core doesn't know the app ID it should ask HMI for a policy update.
+     1. Alternatively HMI can call `UpdateSDL` to enforce SDL to peform the policy update.
+2. Policy HMI impl. (incl. SYNCP) should have the ability to use embedded modem to send policy snapshot to the policy server.
+3. All CloudAppProperties RPCs and related structs should also be available through HMI_API
+   1. This allows OEM stores easier communication with SDL.
+4. A new enum called "CoreTransportRole" with "CLIENT" and "SERVER" elements should be added
+   1. CloudAppProperties should have a parameter for "coreTransportRole"
+
+### Launching an app (alternative solution)
+
+If the user taps on an app icon and
+  a) and the app is listed as per `UpdateAppList` (app is set with `enabled=true`)
+    1. HMI should use `OnAppActivated` to Core to trigger app activation
+    2. Core recognizes the app is not connected. Core sends `ActivateApp` back to HMI
+    3. HMI launches the web engine to load the web app
+  b) and the app is listed by the HMI directly (app is set with `enabled=false`)
+    The main proposal flow applies (see section 1.2)
+
+### Downsides and difficulties
+
+There are several difficulties to deal with if the transport adapter concept is applied to web apps:
+
+The `endpoint` parameter could not be used as a launch command (`/someApp/./index.html`) to launch a local web app unless SDL Core has full control of the WebEngine. There is a potential risk that Core launches an app which is already launched but not connected to SDL. This leads to multiple instances of a single app. Launching an app should be in the responsibility of the HMI. 
+
+The `enabled` parameter is implemented slightly different as expected. If set to `true` is treats the app as a cloud app. It cannot be used to enable or disable an app as the name would imply. Changing `enabled` flag behavior may be a breaking change that (in worse case) require versioning at the policy server.
+
+### Add `LocalAppProperties`
+
+Another solution would be to add another set of RPCs only for local web apps. These RPCs would be tailored to the web apps and would not break any existing behavior for cloud apps.
+
+### Conclusion
+
+Many difficulties with remote cloud apps don't apply to local (web) apps. All the effort to make cloud (or local) app properties to work has only little benefit. SDL Core would already know about apps installed and *could*  update the policies before they are launched. Due to the fact that the IVI and the HMI already know everything about the local web apps, and that it can easily launch such apps doesn't require app properties on the SDL side.
```

Comment:
```suggestion
Many difficulties with remote cloud apps don't apply to local (web) apps. All the effort to make cloud (or local) app properties to work has only little benefit. SDL Core would already know about apps installed and *could* update the policies before they are launched. Due to the fact that the IVI and the HMI already know everything about the local web apps, and that it can easily launch, such apps doesn't require app properties on the SDL side.
```